products/activedirectory/terraform.yaml

```diff
@@ -0,0 +1,32 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Domain: !ruby/object:Overrides::Terraform::ResourceOverride
+    id_format: "{{name}}"
+    import_format: ["{{name}}"]
```

Comment:
I think import isn't actually going to work for people, and this is probably a bug in other resources too that we just haven't caught yet. Since we don't set domain_name on read, import won't set it. However, the field is Required, so the user has to set it in their config. This will show a diff.

One thing that may work is setting the import_format to the long form of the name: `projects/{project}/locations/global/domains/{domainName}`. Give that a try first. If it doesn't work, you might need to set a custom import function.

You'll probably want to test this out by building the provider instead of with the test, because the test does ImportStateVerifyIgnore for anything that's url_param_only.

Reply:
I am doing custom_import for `name` and tested the import command with provider binary

```
custom_code: !ruby/object:Provider::Terraform::CustomCode
   custom_import: templates/terraform/custom_import/self_link_as_name.erb 
```
```
  terraform import google_active_directory_domain.ad-domain projects/xxx-external-org/locations/global/domains/mydomain.org1.com
```